src/main.py

```diff
@@ -32,10 +34,29 @@ def calculate_scaled_mouse_position(position: tuple[int, int]) -> tuple[int, int
 
         self.app_state_manager.ui_manager.calculate_scaled_mouse_position = calculate_scaled_mouse_position
 
+    def run_debug(self) -> None:
+        surface = virtual_screen
+        font = pygame.font.SysFont("Comic Sans", 18)
+        black = (0, 0, 0)
+        current_state_name = self.app_state_manager.active_state.name if self.app_state_manager.active_state else "?"
+        previous_state_name = (
+            self.app_state_manager.previous_state.name if self.app_state_manager.previous_state else "?"
+        )
+        state_display = font.render(
+            f"Game State: {current_state_name}\n" f"Previous State: {previous_state_name}",
+            True,
+            black,
```

Comment:
Nit: booleans with context are harder to read, so let's:

```suggestion
            antialias=True,
            color=black,
```

or define `antialias = True` before this call to feed them both "unnamed"

```Py
        antialias = True
        state_display = font.render(
            f"Game State: {current_state_name}\n" f"Previous State: {previous_state_name}",
            antialias,
            black,
        )
```